packages/ethereum/contracts/HoprChannels.sol

```diff
@@ -523,30 +523,50 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer {
         }
     }
 
+    /**
+     * @dev computes the challenge from the given response.
+     * This is done by misusing ecrecover to perform a ec-point multiplication
+     * with a scalar.
+     */
+    function computeChallenge(bytes32 response) public pure returns (address)  {
```

Comment:
do the singer here?

Reply:
... ? which singer?

Reply:
I guess you meant **signer**, added an explanation in the comment